src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -282,6 +283,29 @@ def _build_os_profile():
 
         return os_profile
 
+    def _build_encryption_settings():
+        if disk_encryption_key_url:
```

Comment:
Do we need this for single pass encryption? or this is for legacy mode?

Reply:
This is for legacy mode